var/spack/repos/builtin/packages/h5bench/package.py

```diff
@@ -58,3 +59,33 @@ def cmake_args(self):
         ]
 
         return args
+
+    @run_after("install")
+    def setup_build_tests(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources(["tests", "samples"])
+
+        launcher = self.mpi_launcher()
+
+        filter_file(r"mpirun", launcher.command, "samples/sync-write-1d-contig-contig.json")
+
```

Comment:
Since these tests can be run in environments without allocations, for example, I recommend adding the `--timeout` option as well.  Assuming the command should normally run in say 2 minutes, you could add the following:

```suggestion
        filter_file(r"-n 2", "-n 2 --timeout=120", "samples/sync-write-1d-contig-contig.json")
```

That way if the package runs say in a CI stack then it will timeout and fail instead of unnecessarily chewing up CI resources.

Reply:
makes sense, done!